src/include/devcomm.h

```diff
@@ -228,10 +239,16 @@ struct ncclChannel {
 };
 static_assert(sizeof(struct ncclChannel) == 0x80*sizeof(int), "ncclChannel must have a pow2 size");
 
+struct scklFlag {
+  uint64_t flag;
+  uint64_t align[3]; // To avoid false sharing
+};
+
 struct ncclDevComm {
   int rank;
   int nRanks;
   int buffSizes[NCCL_NUM_PROTOCOLS];
+  struct scklFlag* scklFlags; // these are used to serialize between thread blocks
```

Comment:
statically allocate this array, if possible

Reply:
Done